### PR TITLE
PDS proxy to appview performance

### PR DESCRIPTION
**Pull request by dholms**
Introduces a temporary measure where a PDS can proxy directly through to the appview without signing a service auth header. This is mean to reduce load on Bluesky PDS instances, and is not intended to be implemented by other Appviews in the network. While this is a bit awkward, it is directionally aligned with possible plans to have the client talk directly to the Appview.

### CODE REVIEW

File: packages/bsky/src/auth-verifier.ts [hunk 1]
```diff
@@ -103,6 +115,17 @@ export class AuthVerifier {
           credentials: { type: 'standard', iss, aud },
         }
       } else if (isBearerToken(ctx.req)) {
+        // @NOTE temporarily accept entryway session tokens to shed load from PDS instances
+        const token = bearerTokenFromReq(ctx.req)
+        const header = token ? jose.decodeProtectedHeader(token) : undefined
+        if (header?.typ === 'at+jwt') {
```
Comment (matthieusieben) [inline packages/bsky/src/auth-verifier.ts:121]: Love it that the `typ` header is coming in handy :-)

File: packages/bsky/tests/entryway-auth.test.ts [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+  const privKeyRaw = await keypair.export()
+  const privKeyEncoded = keyEncoder.encodePrivate(
+    ui8.toString(privKeyRaw, 'hex'),
+    'raw',
+    'pem',
+  )
+  return nodeCrypto.createPrivateKey(privKeyEncoded)
+}
+
+describe('entryway auth', () => {
```
Comment (devinivy) [inline packages/bsky/tests/entryway-auth.test.ts:24]: Testing expiration could be nice too!

**Review by devinivy — APPROVED** [review]

**Review by Vincentroland — APPROVED** [review]

**Review by Vincentroland — APPROVED** [review]

File: packages/bsky/src/auth-verifier.ts (hunk 1 above)
Comment (Vincentroland) [reply packages/bsky/src/auth-verifier.ts:121]: Expilct all

File: packages/bsky/src/auth-verifier.ts [hunk 3]
```diff
@@ -182,6 +205,51 @@ export class AuthVerifier {
...
+          'InvalidToken',
+        )
+      })
+
+    const { sub, aud, scope } = res.payload
+    if (typeof sub !== 'string' || !sub.startsWith('did:')) {
+      throw new AuthRequiredError('Malformed token', 'InvalidToken')
+    } else if (
+      typeof aud !== 'string' ||
+      !aud.startsWith('did:web') ||
```
Comment (matthieusieben) [inline packages/bsky/src/auth-verifier.ts:236]: nit (?) the method name should probably not be alterable

```suggestion
      !aud.startsWith('did:web:') ||
```

**Pull request merged**